### PR TITLE
Add support for returning total results found from find_* funcs

### PR DESCRIPTION
**Pull request by Svampen**
This feature is backwards compatible and will only be used if
total_results is not false or undefined in properties for DocName.
It also requires the adapter(s) to be updated for full effect

I thought I'd create a pull request in case you guys want this in the main repo. It's a variant of #314.
I'm not satisfied with the naming of the type or the props name so I will gladly take suggestions on that :)

### CODE REVIEW

File: src/sumo.erl [hunk 1]
```diff
@@ -229,7 +237,7 @@ delete_all(DocName) ->
   case sumo_store:delete_all(Store, DocName) of
     {ok, NumRows} ->
       _ = case NumRows > 0 of
-        true -> EventId = sumo_event:dispatch(DocName, EventId, deleted_all, []);
+        true -> EventId = sumo_event:dispatch(DocName, EventId, deleted_all,[]);
```
Comment (elvisci) [inline src/sumo.erl]: Missing space right "," on line 240

**Comment by elbrujohalcon** [comment]
As I stated on #314 , I'm not really fond of this feature. I'll let @cabol, @ferigis and others decide, but my vote is to split the concepts.

In any case, for a PR like this one to be merged in this repo you should implement the whole functionality in the default (i.e. mnesia) adaptor and you need to add some tests for it and some generic test cases to the adapter_test_helpers.

**Comment by Svampen** [comment]
If you decided this if of value then I will look into implementing the full functionality for mnesia, though I don't think it supports it in a atomic way (like Riak) so I would probably add a property in the storage_backends to enable it a clarification of inefficiency.

**Comment by cabol** [comment]
Fixed here -> https://github.com/inaka/sumo_db/commit/3c0d4774161499a48e2dc0e3d17afa5461a4dfb1

**Pull request closed without merging**